service-base/src/main/java/org/eclipse/hono/service/plan/PrometheusBasedResourceLimitChecks.java

```diff
@@ -126,47 +143,58 @@ public PrometheusBasedResourceLimitChecks(final WebClient webClient) {
+     * have been reached.
+     * @param tenant The tenant configuration to check the limit against.
+     * @param payloadSize The message payload size in bytes.
+     * @return A future indicating the outcome of the check.
+     *         <p>
+     *         The future will be failed with a {@link ServiceInvocationException}
+     *         if the check could not be performed.
      */
-    public String getHost() {
-        return host;
-    }
+    public Future<Boolean> isMessageLimitReached(final TenantObject tenant,
+            final long payloadSize) {
 
-    /**
-     * Sets the host of the Prometheus server to retrieve metrics from.
-     * <p>
-     * The default value of this property is {@code null}.
-     *
-     * @param host The host name or IP address.
-     */
-    public void setHost(final String host) {
-        this.host = Objects.requireNonNull(host);
-    }
+        Objects.requireNonNull(tenant);
+        final long maxBytes = getMaximumNumberOfBytes(tenant);
+        final LocalDate effectiveSince = getEffectiveSince(tenant);
+        final long periodInDays = getPeriodInDays(tenant);
 
-    /**
-     * Gets the port of the Prometheus server to retrieve metrics from.
-     *
-     * @return port The port number.
-     */
-    public int getPort() {
-        return port;
-    }
+        log.trace("message limit config for tenant [{}] are [{}:{}, {}:{}, {}:{}]", tenant.getTenantId(),
+                FIELD_MAX_BYTES, maxBytes, FIELD_EFFECTIVE_SINCE, effectiveSince, FIELD_PERIOD_IN_DAYS, periodInDays);
 
-    /**
-     * Sets the port of the Prometheus server to retrieve metrics from.
-     * <p>
-     * The default value of this property is 9090.
-     *
-     * @param port The port number.
-     * @throws IllegalArgumentException if the port number is &lt; 0 or &gt; 2^16 - 1
-     */
-    public void setPort(final int port) {
-        if (PortConfigurationHelper.isValidPort(port)) {
-            this.port = port;
+        if (maxBytes == -1 || effectiveSince == null || periodInDays <= 0 || payloadSize <= 0) {
+            return Future.succeededFuture(Boolean.FALSE);
         } else {
-            throw new IllegalArgumentException("invalid port number");
+            final long dataUsagePeriod = calculateDataUsagePeriod(effectiveSince, periodInDays);
+
+            if (dataUsagePeriod <= 0) {
+                return Future.succeededFuture(Boolean.FALSE);
+            }
+
+            final String queryParams = String.format("floor(sum(increase(%s{status=~\"%s|%s\", tenant=\"%s\"} [%sd])))",
+                    MESSAGES_PAYLOAD_SIZE_METRIC_NAME,
+                    MetricsTags.ProcessingOutcome.FORWARDED.asTag().getValue(),
+                    MetricsTags.ProcessingOutcome.UNPROCESSABLE.asTag().getValue(),
+                    tenant.getTenantId(),
+                    dataUsagePeriod);
+            final String cacheId = String.format("%s_bytes_consumed", tenant.getTenantId());
```

Comment:
this is the key, isn't it?

Reply:
yes. Do you suggest renaming it to something like `cacheKey` instead of `cacheId`?

Reply:
Or just `key` ...